docs/content/direct/observability.md

```diff
@@ -0,0 +1,63 @@
+
+### Metrics Endpoint Table
+
+| Controller                      | Protocol | Port  | Path      | AuthN/AuthZ | Notes |
+|----------------------------------|----------|-------|-----------|-------------|-------|
+| kubestellar-controller-manager   | HTTPS    | 8443  | /metrics  | Kubernetes client authentication required when using the Service | Service: `kubestellar-controller-manager-metrics-service` |
+| ks-transport-controller         | HTTP     | 8090  | /metrics  | None (in-cluster) |  |
+| status-addon-controller         | HTTP     | 9280  | /metrics  | None (in-cluster) |  |
+| status-agent-controller         | HTTP     | 8080  | /metrics  | None (in-cluster) |  |
+
+> **Note:** When using the Service for `kubestellar-controller-manager`, protocol is HTTPS and Kubernetes client authentication is required. Direct pod access may use HTTP.
+
+## Debug/Profiling Endpoints
+
+Some KubeStellar components expose Go's built-in pprof debug endpoints for profiling and troubleshooting.
+
+### pprof Endpoint Table
+
+| Controller                      | Protocol | Port  | Path            | AuthN/AuthZ | Notes |
+|----------------------------------|----------|-------|-----------------|-------------|-------|
+| kubestellar-controller-manager   | HTTP     | 8082  | /debug/pprof/   | None        |  |
+| ks-transport-controller         | HTTP     | 8092  | /debug/pprof/   | None        |  |
+| status-addon-controller         | HTTP     | 9282  | /debug/pprof/   | None        |  |
+| status-agent-controller         | HTTP     | 8082  | /debug/pprof/   | None        |  |
+
+## Example: Accessing Metrics and Debug Endpoints
+
+> **Note:** The following example assumes you have a running KubeStellar controller pod and access to the appropriate Kubernetes context and namespace. Adjust the context, namespace, and deployment name as needed for your environment.
+
+```sh
+kubectl --context kind-kubeflex port-forward -n wds1-system deployment/kubestellar-controller-manager 8443:8443 8082:8082
+```
+- Access metrics: [https://localhost:8443/metrics](https://localhost:8443/metrics) (Kubernetes client authentication required)
+- Access pprof: [http://localhost:8082/debug/pprof/](http://localhost:8082/debug/pprof/)
+
+## Grafana Dashboards
+
+- Example Grafana dashboards and configuration can be found in [`monitoring/grafana/`](https://github.com/kubestellar/kubestellar/tree/main/monitoring/grafana).
+- After deploying Prometheus and Grafana (or your preferred stack), you can import dashboards to visualize KubeStellar metrics.
+
+## Additional Resources
+
+- [KubeStellar Monitoring README](https://github.com/kubestellar/kubestellar/tree/main/monitoring/README.md) (one possible way to collect metrics and profiles)
+- [Prometheus Operator Documentation](https://prometheus-operator.dev/) (optional)
+- [Grafana Documentation](https://grafana.com/docs/) (optional)
+- [Go pprof Documentation](https://pkg.go.dev/net/http/pprof) (optional)
```

Comment:
It is not clear what the text "(optional)" is doing on these lines. Do you mean that it is optional for the reader to read these? Optional for the reader to use those? Both are obvious enough without needing any explicit remark.